alf/bin/train.py

```diff
@@ -67,54 +71,149 @@ def _define_flags():
     flags.DEFINE_multi_string('conf_param', None, 'Config binding parameters.')
     flags.DEFINE_bool('store_snapshot', True,
                       'Whether store an ALF snapshot before training')
+    flags.DEFINE_enum('distributed', 'none', ['none', 'multi-gpu'],
+                      'Whether store an ALF snapshot before training')
+    flags.mark_flag_as_required('root_dir')
 
 
 FLAGS = flags.FLAGS
 
 
-@alf.configurable
-def train_eval(root_dir):
-    """Train and evaluate algorithm
+def _setup_logging(rank: int = 0, log_dir: Optional[str] = None):
+    """Setup logging for each process
 
     Args:
-        root_dir (str): directory for saving summary and checkpoints
+        rank: The ID of the process among all of the DDP processes
```

Comment:
For all the args, please add types after the arg names.

Reply:
Done.